File: 1127-last-stone-weight/last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>(Collections.reverseOrder());

        for(int stone:stones){
            maxHeap.add(stone);
        }
        while(maxHeap.size()>1){
            int y = maxHeap.poll();
            int x = maxHeap.poll();
            if(x!=y){
                y = y-x;
                maxHeap.add(y);
            }
        }
        if(maxHeap.isEmpty()){
            return 0;
        }
        return maxHeap.peek();

    }
}

File: 1127-last-stone-weight/last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        heap = []
        for stone in stones:
            heapq.heappush(heap, -stone)
            
        while len(heap)>1:
            y = -heapq.heappop(heap) #this one is bigger
            x = -heapq.heappop(heap)

            if x == y:
                continue
            if x != y:
                 y -= x
                 heapq.heappush(heap, -y)
        return -heapq.heappop(heap) if len(heap) == 1 else 0

            





            
        